pkgs/development/compilers/dtc/default.nix

```
@@ -83,7 +83,7 @@ stdenv.mkDerivation (finalAttrs: {
   mesonAutoFeatures = "auto";
   mesonFlags = [
     (lib.mesonBool "static-build" stdenv.hostPlatform.isStatic)
-    (lib.mesonBool "tests" finalAttrs.doCheck)
+    (lib.mesonBool "tests" (stdenv.buildPlatform.canExecute stdenv.hostPlatform && finalAttrs.doCheck))
```

```suggestion
    (lib.mesonBool "tests" finalAttrs.finalPackage.doCheck)
```

includes the result of canExecute in make-derivation

---

Oh amazing! Thanks for the tip.